DToCToDB.md

```diff
@@ -0,0 +1,1193 @@
+
+
+```python
+# dtoc_11 and dtoc_12, have different column names, therefore correct for this
+dtoc_11 = dtoc_11.rename(columns={'Provider SHA Code':'Provider Parent Org Code', 'Provider SHA Name':'Provider Parent Name'})
+dtoc_12 = dtoc_12.rename(columns={'Provider SHA Code':'Provider Parent Org Code', 'Provider SHA Name':'Provider Parent Name'})
+
+print('dtoc_11\n---')
+dtoc_11.info()
+print('\ndtoc_12\n---')
+dtoc_12.info()
+```
+
+    dtoc_11
+    ---
+    <class 'pandas.core.frame.DataFrame'>
+    Int64Index: 129560 entries, 0 to 16219
+    Data columns (total 16 columns):
+    Year                              129560 non-null int64
+    Month                             129560 non-null int64
+    Provider Parent Org Code          129560 non-null object
+    Provider Parent Name              129560 non-null object
+    Provider Org Code                 129560 non-null object
+    Provider Org Name                 129560 non-null object
+    Local Authority Code              129560 non-null object
+    Local Authority Name              129560 non-null object
+    Acute Or Non Acute Description    129560 non-null object
+    Reason For Delay                  129560 non-null object
+    NHS A SUM                         129547 non-null float64
+    NHS B SUM                         129534 non-null float64
+    Social Care A SUM                 129541 non-null float64
+    Social Care B SUM                 129528 non-null float64
+    Both A SUM                        129560 non-null int64
+    Both B SUM                        129560 non-null int64
+    dtypes: float64(4), int64(4), object(8)
+    memory usage: 16.8+ MB
+    
+    dtoc_12
+    ---
+    <class 'pandas.core.frame.DataFrame'>
+    Int64Index: 176060 entries, 0 to 15919
+    Data columns (total 16 columns):
+    Year                              176060 non-null int64
+    Month                             176060 non-null int64
+    Provider Parent Org Code          176060 non-null object
+    Provider Parent Name              176060 non-null object
+    Provider Org Code                 176060 non-null object
+    Provider Org Name                 176060 non-null object
+    Local Authority Code              176060 non-null object
+    Local Authority Name              176060 non-null object
+    Acute Or Non Acute Description    176060 non-null object
+    Reason For Delay                  176060 non-null object
+    NHS A SUM                         176051 non-null float64
+    NHS B SUM                         176044 non-null float64
+    Social Care A SUM                 176052 non-null float64
+    Social Care B SUM                 176048 non-null float64
+    Both A SUM                        176060 non-null int64
+    Both B SUM                        176059 non-null float64
+    dtypes: float64(5), int64(3), object(8)
+    memory usage: 22.8+ MB
+
+
```

Comment:
High level, you could centralize all the main logic of downloading + concatenating + renaming + changing the months into a single function.  Pseudocode:

```
def download(urls, header_nums):
    frames = []
    for i, url in urls.enumerate():
        header_num = header_nums[i]
        download + append to frames, using the right header_num
    concatenate
    do the renaming
    replace months
```

Then you can just do this:

```
years = range(2011,2017)
urls = [[...],[...],[...],...]
header_nums = [[...],[...],...]
frames =[]
for year in years:
    frames.append(download(urls, header_nums))